infrastructure/redeploy-operator.yaml

```diff
@@ -0,0 +1,135 @@
+
+            if [ -n "$FAILED_PRODUCTION_DEPLOYMENTS" ]; then
+              echo "Redeploying failed production environments"
+
+              for deployment in $FAILED_PRODUCTION_DEPLOYMENTS; do
+                if [[ $deployment =~ "dpl-cms" ]]; then
+                  continue
+                fi
+                if [[ $deployment =~ "dpl-bnf" ]]; then
+                  continue
+                fi
+                echo "$deployment: deploying"
+                /lagoon deploy latest -p "$deployment" -e "main" --force
+              done
+            else
+              echo "No failed production deployments to redeploy"
+            fi
+
+            echo "Now checking for failed development site deployments"
+            FAILED_DEVELOPMENT_DEPLOYMENTS=$(/lagoon raw --raw "query allProjects {
+              allProjects {
+              name
+                environments(type: DEVELOPMENT) {
+                  name
+                  deployments(limit: 1) {
+                    status
+                    created
+                  }
+                }
+              }
+            }" | jq -r '.allProjects[] | .name as $name | .environments[].deployments[] | select(.status == "failed") | ($name)')
+
+            if [ -n "$FAILED_DEVELOPMENT_DEPLOYMENTS" ]; then
+              echo "Redeploying failed development environments"
+
+              for deployment in $FAILED_DEVELOPMENT_DEPLOYMENTS; do
+                if [[ $deployment =~ "dpl-cms" ]]; then
+                  continue
+                fi
+                if [[ $deployment =~ "dpl-bnf" ]]; then
+                  continue
+                fi
+                echo "$deployment: deploying"
+                /lagoon deploy latest -p "$deployment" -e "main" --force
+              done
+            else
+              echo "No failed development deployments to redeploy"
+            fi
+          }
+
+          echo "start redeploying"
+
+          while [ true ]
+          do
+            redeployDeployments
+            echo "waiting for 5 minutes before checking for failed deployments again"
+            sleep 300
+          done
+
+      volumeMounts:
+        - name: lagoon-known-hosts
+          mountPath: /root/.ssh/known_hosts
+          subPath: known_hosts
+        - name: lagoon-config-volume
+          mountPath: /config
+        - name: writable-lagoon-config
+          mountPath: /root/config
+        - name: lagoon-ssh-key-volume
+          mountPath: /root/.ssh/id_rsa
+          subPath: id_rsa
+          readOnly: true
+  volumes:
+    - name: lagoon-config-volume
+      configMap:
+        name: lagoon-config
+    - name: writable-lagoon-config
+      emptyDir:
+        medium: Memory
+    - name: lagoon-known-hosts
+      configMap:
+        name: lagoon-known-hosts
+    - name: lagoon-ssh-key-volume
+      secret:
+        secretName: lagoon-ssh-key
+  restartPolicy: Never
```

Comment:
This is how you tested it before the new image, right? Seems unrelated to the change of getting to create the image

Reply:
no this is supposedly how i will use the image to make a deployment

Reply:
Ok, so... not that related to the new image you are adding?